lib/core/mock/identity-group.service.mock.ts

```diff
@@ -15,104 +15,131 @@
  * limitations under the License.
  */
 
-import { IdentityGroupModel, IdentityGroupCountModel } from '../models/identity-group.model';
+import { Injectable } from '@angular/core';
+import { mockIdentityGroups, mockIdentityRoles } from './identity-group.mock';
+import { Observable, of } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { IdentityGroupServiceInterface } from '../services/identity-group.interface';
+import {
+    IdentityGroupModel,
+    IdentityGroupQueryResponse,
+    IdentityGroupQueryCloudRequestModel,
+    IdentityGroupSearchParam,
+    IdentityGroupCountModel
+} from '../models/identity-group.model';
 import { IdentityRoleModel } from '../models/identity-role.model';
 
-export const mockIdentityGroup1 = <IdentityGroupModel> {
-    id: 'mock-group-id-1', name: 'Mock Group 1', path: '/mock', subGroups: []
-};
+Injectable({ providedIn: 'root' });
+export class IdentityGroupServiceMock implements IdentityGroupServiceInterface {
 
-export const mockIdentityGroup2 = <IdentityGroupModel> {
-    id: 'mock-group-id-2', name: 'Mock Group 2', path: '', subGroups: []
-};
+    getGroups(): Observable<IdentityGroupModel[]> {
+        return of(mockIdentityGroups);
+    }
 
-export const mockIdentityGroup3 = <IdentityGroupModel> {
-  id: 'mock-group-id-3', name: 'Mock Group 3', path: '', subGroups: []
-};
+    getAvailableRoles(_groupId: string): Observable<IdentityRoleModel[]> {
+        return of(mockIdentityRoles);
+    }
 
-export const mockIdentityGroup4 = <IdentityGroupModel> {
-    id: 'mock-group-id-4', name: 'Mock Group 4', path: '', subGroups: []
-};
+    getAssignedRoles(_groupId: string): Observable<IdentityRoleModel[]> {
+        return of(mockIdentityRoles);
+    }
 
-export const mockIdentityGroup5 = <IdentityGroupModel> {
-    id: 'mock-group-id-5', name: 'Mock Group 5', path: '', subGroups: []
-};
+    assignRoles(_groupId: string, _roles: IdentityRoleModel[]): Observable<any> {
+        return of();
+    }
 
-export const mockIdentityGroupsCount = <IdentityGroupCountModel> { count: 10 };
+    removeRoles(_groupId: string, _roles: IdentityRoleModel[]): Observable<any> {
+        return of();
+    }
 
-export const mockIdentityGroups = [
-    mockIdentityGroup1, mockIdentityGroup2, mockIdentityGroup3, mockIdentityGroup4, mockIdentityGroup5
-];
+    getEffectiveRoles(_groupId: string): Observable<IdentityRoleModel[]> {
+        return of(mockIdentityRoles);
+    }
 
-export const mockApplicationDetails = {id: 'mock-app-id', name: 'mock-app-name'};
+    queryGroups(_requestQuery: IdentityGroupQueryCloudRequestModel): Observable<IdentityGroupQueryResponse> {
+        return of();
+    }
 
-export const roleMappingMock = [
-    { id: 'role-id-1', name: 'role-name-1' }, { id: 'role-id-2', name: 'role-name-2' }
-];
+    getTotalGroupsCount(): Observable<IdentityGroupCountModel> {
+        return of({ count: mockIdentityGroups.length });
```

Comment:
we need it, it used in unit testing https://github.com/Alfresco/alfresco-ng2-components/blob/5205628a0e146698913492e55876d24d8c71cbba/lib/core/services/identity-group.service.spec.ts#L273

Reply:
changed to use `mockIdentityGroupsCount `, because it also used [here](https://github.com/Alfresco/alfresco-ng2-components/blob/7148d299a8acc54994710a73285d362929c7998a/lib/core/services/identity-group.service.spec.ts#L271).